### PR TITLE
[IMP] hr_expense: add attachments from expense to journal entries

### PR DESCRIPTION
**Pull request by alialfie**
This commit adds a copy of the attachments from the expense report to the created journal entry.

task-3443042

### CODE REVIEW

**Comment by rugo-odoo** [comment]
LGTM

**Review by smetl — CHANGES_REQUESTED** [review]

File: addons/hr_expense/models/hr_expense.py [hunk 1]
```diff
@@ -1507,7 +1508,8 @@ def _prepare_bills_vals(self):
             'move_type': 'in_invoice',
             'partner_id': self.employee_id.sudo().work_contact_id.id,
             'currency_id': self.currency_id.id,
-            'line_ids':[Command.create(expense._prepare_move_lines_vals()) for expense in self.expense_line_ids],
+            'line_ids': [Command.create(expense._prepare_move_lines_vals()) for expense in self.expense_line_ids],
+            'attachment_ids': [Command.set([attachment.copy({'res_model': 'account.move', 'res_id': False}).id for attachment in self.expense_line_ids.message_main_attachment_id])],
```
[the remark this answers is not in the document]
Comment (smetl) [reply addons/hr_expense/models/hr_expense.py:1512]: Or use `Command.create(attachment.copy_data({'res_model': 'account.move', 'res_id': False})[0])` ?

**Comment by alialfie** [comment]
@rugo-odoo  @smetl Hi, new changes are added with `copy_data` instead of `copy` and tests. Thanks!

File: addons/hr_expense/models/hr_expense.py [hunk 2]
```diff
@@ -713,6 +713,9 @@ def _prepare_payments_vals(self):
             'payment_method_line_id': payment_method_line.id,
             'currency_id': self.currency_id.id,
             'line_ids': [Command.create(line) for line in move_lines],
+            'attachment_ids': [
+                Command.create(attachment.copy_data({'res_model': 'account.move', 'res_id': False, 'raw': attachment.raw})[0])
+                for attachment in self.message_main_attachment_id],
```
[the remark this answers is not in the document]
Comment (alialfie) [reply addons/hr_expense/models/hr_expense.py]: done!

**Comment by qdp-odoo** [comment]
@robodoo r+

:+1:

**Pull request closed without merging**